docs/proposals/zeroconf.md

```diff
@@ -0,0 +1,142 @@
+
+## Outstanding Questions
+
++ What would a generic Akri ZeroConf Broker do? In practice, the application developer would likely wish to implement the Broker for their specific application.
+
+The limit of a generic Akri ZeroConf Broker is to enumerate services that it discovers and this is demonstrated by a sample Broker included in the ZeroConf Protocol implementation. In practice, a ZeroConf Broker would need to be aware of the implementation(s) of the service that it "twins". There is a limitation on the transport protocols that are accessible to Akri (discussed below) but, for the permitted transport protocols, there are potentially limitless service types and implementation details and these are all potentially accessible to an Akri Broker using this ZeroConf protocol implementation.
+
++ How to treat support for Kubernetes-supporting service types (TCP, UDP, SCTP)?
+
+Kubernetes supports [TCP, UDP, SCTP](https://kubernetes.io/docs/concepts/services-networking/service/#protocol-support) transport protocols. Service discovery supports other transport protocols. Because Akri is dependent on Kubernetes, the Akri ZeroConf Protocol implementation only supports these 3 transport protocols too.
+
+## Feature Requests
+
++ Discovery should differentiate between services that are supportable (!) by Kubernetes (TCP, UDP, SCTP) and those that aren't
+
+The [`zeroconf-filter`](https://github.com/DazWilkin/akri-pest) crate used by the ZeroConf Protocol implementation only permits: TCP, UDP and SCTP.
+
++ Discovery should apply user-defined filters on Services so that the Agent only attempts to discover filtered services
+
+The [`zeroconf-filter`](https://github.com/DazWilkin/akri-pest) crate used by the ZeroConf Protocol implementation enables filtering of discovered services. The filter permits `name`, `domain`, `kind` and `port`filtering. However, the [`zeroconf`] crate only supports `kind` filtering and so other terms would be ignored.
+
+The `filter` is specific in the ZeroConf CRD Configuration:
+
+```YAML
+properties:
+  zeroconf: # {{ZeroConfDiscoveryHandler}}
+    type: object
+    properties:
+      filter: 
+        type: string
+```
+
+And thus, the Agent may be applied to the cluster:
+
+```YAML
+apiVersion: akri.sh/v0
+kind: Configuration
+metadata:
+  name: zeroconf
+spec:
+  protocol:
+    zeroconf:
+      filter: 'kind="_http._tcp"'
+```
+
+## References
+
++ [Zero-configuration networking](https://en.wikipedia.org/wiki/Zero-configuration_networking).
++ [IANA ZeroConf Service Name and Transport Protocol Port Number Registry](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?skey=9&page=132)
++ [Rust `zeroconf` crate](https://crates.io/crates/zeroconf)
++ [Development Branch of Akri ZeroConf Protocol & Broker](https://github.com/DazWilkin/akri/tree/protocol-zeroconf)
```

Comment:
Are you thinking we leave this reference in until we have Zeroconf merged into main?

Reply:
I don't know how to represent this type of change.

Reply:
We can leave it as is for the time being